plugins/modules/monitoring/pagerduty_change.py

```diff
@@ -0,0 +1,189 @@
+    revision: '4.2'
+    environment: production
+    link_url: https://github.com/ansible-collections/community.general/pull/1269
+    link_text: View changes on GitHub
+'''
+
+from ansible.module_utils.urls import fetch_url
+from ansible.module_utils.basic import AnsibleModule
+from datetime import datetime
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            integration_key=dict(required=True, type='str'),
+            summary=dict(required=True, type='str'),
+            source=dict(required=False, default='Ansible', type='str'),
+            user=dict(required=False, type='str'),
+            repo=dict(required=False, type='str'),
+            revision=dict(required=False, type='str'),
+            environment=dict(required=False, type='str'),
+            link_url=dict(required=False, type='str'),
+            link_text=dict(required=False, type='str'),
+            url=dict(required=False,
+                     default='https://events.pagerduty.com/v2/change/enqueue', type='str'),
+            validate_certs=dict(default=True, type='bool')
+        ),
+        supports_check_mode=True
+    )
+
+    # API documented at https://developer.pagerduty.com/docs/events-api-v2/send-change-events/
+
+    url = module.params['url']
+    headers = {'Content-Type': 'application/json'}
+
+    if module.check_mode:
+        _response, info = fetch_url(
+            module, url, headers=headers, method='POST')
+
+        if info['status'] == 400:
+            module.exit_json(changed=True)
+        else:
+            module.fail_json(
+                msg='Checking the PagerDuty change event API returned an unexpected response: %d' % (info['status']))
+
+    custom_details = {}
+
+    if module.params['user']:
+        custom_details['user'] = module.params['user']
+
+    if module.params['repo']:
+        custom_details['repo'] = module.params['repo']
+
+    if module.params['revision']:
+        custom_details['revision'] = module.params['revision']
+
+    if module.params['environment']:
+        custom_details['environment'] = module.params['environment']
+
+    now = datetime.utcnow()
+    timestamp = now.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+
+    payload = {
+        'summary': module.params['summary'],
+        'source': module.params['source'],
+        'timestamp': timestamp,
+        'custom_details': custom_details
+    }
+
+    event = {
+        'routing_key': module.params['integration_key'],
+        'payload': payload
+    }
+
+    if module.params['link_url']:
+        link = {
+            'href': module.params['link_url']
+        }
+
+        if module.params['link_text']:
+            link['text'] = module.params['link_text']
+
+        event['links'] = [link]
+
+    _response, info = fetch_url(
+        module, url, data=module.jsonify(event), headers=headers, method='POST')
+
+    if info['status'] == 202:
+        module.exit_json(changed=True)
+    else:
+        module.fail_json(
```

Comment:
In case the event already exists, shouldn't it be updated according to possibly changed parameters?

Users expect that if they execute an Ansible module twice with exactly the same parameters, that on the second run they get `changed=False`. (Assuming the parameters do not contain something like `force=true` or other values that do not work with idempotency, like relative timestamps.)

Reply:
Hmm... I'm not sure if this is possible in this case. The purpose of PagerDuty's change events feature is to track each time a deploy (or other change) is made. If this module was set up such that it resulted in the same event being created each time, then I think a user would expect to see a new event in PagerDuty each time.

If they were to include something that changed with each deploy, like a revision or SHA, then could possibly work, but comes with its own tradeoffs. For one, I don't think there is a way to know that an event was already tracked for that revision. Also, if you needed to redeploy the same SHA again for some reason, then I would expect to see different events in PagerDuty for each deploy.

Reply:
Ah, so do I understand it correctly that there's no sense in actually getting and modifying events? In that case, I guess the current behavior is ok. I guess it wouldn't hurt to extend the module's description so that it is also clear to users. (I hope they already know, but maybe they don't :) )

Reply:
Correct, the is currently no mechanism available to get and modify an event, and I think for the use case we are targeting it wouldn't be desired. I'll update the documentation accordingly.